ToolBox/FxcmDownloader/FxcmDataDownloader.cs

```diff
@@ -163,35 +161,39 @@ public IEnumerable<BaseData> Get(Symbol symbol, Resolution resolution, DateTime
             // initialize session
             RequestTradingSessionStatus();
 
-            Console.WriteLine("Downloading data from {0} to {1}...", startUtc.ToShortDateString(), endUtc.ToShortDateString());
+            Console.WriteLine("Downloading {0} data from {1} to {2}...", Enum.GetName(typeof(Resolution), resolution), startUtc.ToShortDateString(), endUtc.ToShortDateString());
 
-            // download bars
-            var totalBars = new List<TradeBar>();
+            //Find best FXCM  paramrs
+            IFXCMTimingInterval interval = ToFXCMInterval(resolution);
+            
 
-            // calculate the maximum time span for one request (using 10-second bars)
-            const int maxBarsPerRequest = 300;
-            var timeSpanPerRequest = TimeSpan.FromSeconds(maxBarsPerRequest * 10);
+            // download data
+            var totalBaseData = new List<BaseData>();
 
-            var start = startUtc;
-            var end = startUtc + timeSpanPerRequest;
+            //var start = startUtc;
+            var end = endUtc;
 
```

Comment:
If endUtc does not have time components(only date, as it is now), this would start requesting data for the previous day (excluding the last date of the requested interval). I would add .AddDays(1) maybe not here but in the calling program, so this method can also be used to request an intraday period (such as a single hour).